src/test/sigopcount_tests.cpp

```diff
@@ -64,4 +65,16 @@ BOOST_AUTO_TEST_CASE(GetSigOpCount)
     BOOST_CHECK_EQUAL(p2sh.GetSigOpCount(scriptSig2), 3U);
 }
 
+
+BOOST_AUTO_TEST_CASE(blockSigOpAcceptLimit)
+{
+    BOOST_CHECK_EQUAL(Policy::blockSigOpAcceptLimit(0), 20000);
+    BOOST_CHECK_EQUAL(Policy::blockSigOpAcceptLimit(100), 20000);
+    BOOST_CHECK_EQUAL(Policy::blockSigOpAcceptLimit(50000), 20000);
+    BOOST_CHECK_EQUAL(Policy::blockSigOpAcceptLimit(1e6), 20000);
+    BOOST_CHECK_EQUAL(Policy::blockSigOpAcceptLimit(1e6 + 1), 40000);
+    BOOST_CHECK_EQUAL(Policy::blockSigOpAcceptLimit(2e6), 40000);
+    BOOST_CHECK_EQUAL(Policy::blockSigOpAcceptLimit(4e6 + 1), 100000);
```

Comment:
I updated the test values to not use 1e6 notation. 

> I would test for a set of X1000000, X1000000+1 and X*1000000-1 for various values of X

I added these tests for X values of 1 and 2.

> typical edges cases values (0, 1, uint.max)

I added these tests as well.

> table driven tests

I looked into achieving this by simply using a std::map, and iterating over the map. I think in this case it made the test less legible (since it added lines of code and increased cognitive load on the reader), and made test failures harder to find (because the exact line number wasn't specified). I think this would make more sense when the function being tested is more complex, or if we had a better framework for doing data-driven tests.